x-pack/plugins/ingest_manager/server/services/epm/elasticsearch/transform/install.ts

```diff
@@ -0,0 +1,174 @@
+
+  const transformPathDatasets = datasets.reduce<TransformPathDataset[]>((acc, dataset) => {
+    transformPaths.forEach((path) => {
+      if (isDatasetTransform(path, dataset.path)) {
+        acc.push({ path, dataset });
+      }
+    });
+    return acc;
+  }, []);
+
+  const transformRefs = transformPathDatasets.reduce<EsAssetReference[]>(
+    (acc, transformPathDataset) => {
+      if (transformPathDataset) {
+        acc.push({
+          id: getTransformNameForInstallation(transformPathDataset, installNameSuffix),
+          type: ElasticsearchAssetType.transform,
+        });
+      }
+      return acc;
+    },
+    []
+  );
+
+  // get and save transform refs before installing transforms
+  await saveInstalledEsRefs(savedObjectsClient, registryPackage.name, transformRefs);
+
+  const transforms: TransformInstallation[] = transformPathDatasets.map(
+    (transformPathDataset: TransformPathDataset) => {
+      return {
+        installationName: getTransformNameForInstallation(transformPathDataset, installNameSuffix),
+        content: getAsset(transformPathDataset.path).toString('utf-8'),
+      };
+    }
+  );
+
+  const installationPromises = transforms.map(async (transform) => {
+    return installTransform({ callCluster, transform });
+  });
+
+  const installedTransforms = await Promise.all([
+    Promise.all(installationPromises),
+  ]).then((results) => results.flat());
+
+  const currentInstallation = await getInstallation({
+    savedObjectsClient,
+    pkgName: registryPackage.name,
+  });
+
+  // remove the saved object reference
+  await deleteTransformRefs(
+    savedObjectsClient,
+    currentInstallation?.installed_es || [],
+    registryPackage.name,
+    previousInstalledTransformEsAssets.map((asset) => asset.id),
+    installedTransforms.map((installed) => installed.id)
+  );
+
+  return installedTransforms;
+};
+
+const isTransform = (path: string) => {
+  const pathParts = Registry.pathParts(path);
+  return pathParts.type === ElasticsearchAssetType.transform;
+};
+
+const isDatasetTransform = (path: string, datasetName: string) => {
+  const pathParts = Registry.pathParts(path);
+  return (
+    !path.endsWith('/') &&
+    pathParts.type === ElasticsearchAssetType.transform &&
+    pathParts.dataset !== undefined &&
+    datasetName === pathParts.dataset
+  );
+};
+
+async function installTransform({
+  callCluster,
+  transform,
+}: {
+  callCluster: CallESAsCurrentUser;
+  transform: TransformInstallation;
+}): Promise<EsAssetReference> {
+  const callClusterParams: {
+    method: string;
+    path: string;
+    query: string;
+    ignore?: number[];
+    body: any;
+    headers?: any;
+  } = {
+    method: 'PUT',
+    path: `/_transform/${transform.installationName}`,
+    query: 'defer_validation=true',
+    body: transform.content,
+  };
+
+  await callCluster('transport.request', {
+    method: 'DELETE',
+    query: 'force=true',
+    path: `_transform/${transform.installationName}`,
```

Comment:
All the other paths start with a `/` but not here. Is this on purpose?

Reply:
rectified this, it did not make a difference, but good to be consistent.